x-pack/plugins/reporting/public/index.ts

```diff
@@ -5,22 +5,41 @@
  * 2.0.
  */
 
-import { PluginInitializerContext } from 'src/core/public';
-import { ReportingAPIClient } from './lib/reporting_api_client';
+import type { PluginInitializerContext } from 'src/core/public';
 import { ReportingPublicPlugin } from './plugin';
-import { getSharedComponents } from './shared';
+import type { ReportingPublicComponents } from './shared/get_shared_components';
 
+/**
+ * Setup contract for the Reporting plugin.
+ */
 export interface ReportingSetup {
+  /**
+   * Used to inform plugins if Reporting config is compatible with UI Capabilities / Application Sub-Feature Controls
```

Comment:
Do you know whether our generated docs interpret the `{@link }` inline tag? Maybe we can `{@link Capabilities}` to core's interface?

Reply:
It seems our system for automated API documentation is going through some re-tooling right now. I'd like to consider this PR just the "first" serious effort into having any kind of documentation. Coming soon, we'll have some kind of faster system for improving the documentation and things like this will be easier to figure out.